apps/meteor/app/api/server/v1/moderation.ts

```diff
@@ -113,6 +168,69 @@ API.v1.addRoute(
 	},
 );
 
+API.v1.addRoute(
+	'moderation.user.reportsByUserId',
+	{
+		authRequired: true,
+		validateParams: isGetUserReportsParams,
+		permissionsRequired: ['view-moderation-console'],
+	},
+	{
+		async get() {
+			try {
+				const { userId, selector = '' } = this.queryParams;
+				const { sort } = await this.parseJsonQuery();
+				const { count = 50, offset = 0 } = await getPaginationItems(this.queryParams);
+
+				const user = await Users.findOneById<IUser>(userId, {
+					projection: {
+						_id: 1,
+						username: 1,
+						name: 1,
+						avatarETag: 1,
+						active: 1,
+						roles: 1,
+						emails: 1,
+						createdAt: 1,
+					},
+				});
+
+				const escapedSelector = escapeRegExp(selector);
+				const { cursor, totalCount } = ModerationReports.findUserReportsByReportedUserId(userId, escapedSelector, {
+					offset,
+					count,
+					sort,
+				});
+
+				const [reports, total] = await Promise.all([cursor.toArray(), totalCount]);
+
+				const emailSet = new Map<IUserEmail['address'], IUserEmail>();
+
+				reports.forEach((report) => {
+					const email = report.reportedUser?.emails?.[0];
+					if (email) {
+						emailSet.set(email.address, email);
+					}
+				});
+				if (user) {
+					user.emails = Array.from(emailSet.values());
+				}
+
+				return API.v1.success({
+					user,
+					reports,
+					count: reports.length,
+					total,
+					offset,
+				});
+			} catch (error) {
+				SystemLogger.error(error);
+				return API.v1.failure('Error while fetching user reports');
+			}
```

Comment:
same as the previous one: this try/catch isn't really needed, since the API code has one already for each endpoint.

Reply:
Thank you, removed!